oscplot.c

```diff
@@ -3766,8 +3766,11 @@ static void device_list_treeview_init(OscPlot *plot)
 		plot_channels_add_device(plot, dev_name);
 		priv->nb_input_devices++;
 
-		for (j = 0; j < nb_channels; j++) {
-			struct iio_channel *ch = iio_device_get_channel(dev, j);
+		GArray *channels = get_iio_channels_naturally_sorted(dev);
+
+		for (j = 0; j < channels->len; ++j) {
+			struct iio_channel *ch = g_array_index(channels,
+				struct iio_channel *, j);
```

Comment:
this needs to update an error path, otherwise it leaks;
```
			pic = plot_iio_channel_new(priv->ctx);
			if (!pic)
				return;
```
should be   `break` instead of `return`

Reply:
good catch